crypto/s2n_certificate.c

```diff
@@ -339,6 +347,12 @@ int s2n_cert_chain_and_key_load_pem(struct s2n_cert_chain_and_key *chain_and_key
     /* Validate the leaf cert's public key matches the provided private key */
     POSIX_GUARD(s2n_pkey_match(&public_key, chain_and_key->private_key));
 
+    /* Store public key at cert_chain to cut down expensive use of s2n_asn1der_to_public_key_and_type */
+    POSIX_GUARD(s2n_cert_set_public_key(chain_and_key->cert_chain->head, public_key));
+
+    /* Reset the old struct, so we don't clean up chain_and_key->cert_chain->head */
+    s2n_pkey_zero_init(&public_key);
```

Comment:
Can you use the named macro we created for this model? https://github.com/aws/s2n-tls/blob/a862c877a9e101d010c08375675897afa50f7a89/utils/s2n_safety.h#L76-L84

Reply:
Done